### ISSUE TITLE
XF style layout

### ISSUE THREAD

**Issue by enkiv2** [NaNoGenMo/2022#6]
I'm sort of drawing a blank for content-related ideas this year, but I recently read the very visually interesting Xenofeminist Manifesto and I'd like to see if I can write something to duplicate that layout (probably on Society of the Spectacle).

**Comment by enkiv2** [comment]
Source: https://github.com/enkiv2/misc/blob/master/nanogenmo-2022/XFify.py
Generated output: https://www.lord-enki.net/society-of-the-spectacle.pdf

The original Society of the Spectacle is only 29k words, but since I am turning every letter of every section header (and the first letter of every section) into an image, and we generally respect the conversion rate of one picture per one thousand words, this output *probably* counts as complete -- especially seeing as how this is now 946 pages long. Alternatively, since I laid out each page using image manipulation, each page might count as an image, for a word count equivalent of 946k!

**Comment by enkiv2** [comment]
For those who don't want to open up a giant PDF, this is what my script mostly does:

![The letter X superimposed on a rococo painting tinted red](https://github.com/enkiv2/misc/blob/master/nanogenmo-2022/merged.png?raw=true)

It mimics the XenoFeminist Manifesto's remarkable section headers, which render huge half-page single letters as specific works of art run though a filter that looks like the output of a risograph machine:

![Some section headers from XF](https://cdn-ed.versobooks.com/images/000014/489/XF_blog_6-7945c37eb44ac2384be0c7e9368054ed.jpg)

Basically, what I do is I take a randomly selected image from my giant image directory, convert it to greyscale, and make two versions -- one where it has been multiplied with a flat color, and another where it has been inverted, multiplied with the inverse of that flat color, then inverted again. The result is two versions of the image -- one where the tint level is proportional to the lightness of the pixel, and one where it is inversely proportional. I use a rendered character to mask one, then put it on top of the other.

**Comment by enkiv2** [comment]
To run for yourself:

> cat society-of-the-spectacle.md | ./XFify.py ~/Pictures/ ; img2pdf page*.png --output society-of-the-spectacle.pdf

**Comment by lizadaly** [comment]
This rules.

**Comment by enkiv2** [comment]
Thanks!

**Comment by enkiv2** [comment]
I have an improved version: https://www.lord-enki.net/society-of-the-spectacle-2.pdf

I've started to distinguish between (titled) sections and (numbered) sub-sections, stripped smartquotes from section headers, used the color schemes for the entire sections, centered the letters in section headers vertically and horizontally, and forced spaces in section headers to correspond with page breaks. The resulting file is only 216 pages long, and only 183mb.
